src/runtime/storageNG/tests/store-sequence-test.ts

```diff
@@ -0,0 +1,170 @@
+import {CRDTCountTypeRecord, CRDTCount, CountOpTypes, CountData} from '../../crdt/crdt-count.js';
+import {DriverFactory, Driver, ReceiveMethod, StorageDriverProvider, Exists} from '../drivers/driver-factory.js';
+
+class MockDriver<Data> extends Driver<Data> {
+  receiver: ReceiveMethod<Data>;
+  async read(key: string) { throw new Error("unimplemented"); }
+  async write(key: string, value: {}) { throw new Error("unimplemented"); }
+  registerReceiver(receiver: ReceiveMethod<Data>) {
+    this.receiver = receiver;
+  }
+  async send(model: Data): Promise<boolean> {
+    return true;
+  }
+}
+
+class MockStorageDriverProvider implements StorageDriverProvider {
+  willSupport(storageKey: string) {
+    return true;
+  }
+  driver<Data>(storageKey: string, exists: Exists): Driver<Data> {
+    return new MockDriver<Data>(storageKey, exists);
+  }
+}
+
+describe('Store Flow', async () => {
+  // Tests a model resync request happening synchronously with model updates from the driver
+  it('services a model request and applies 2 models', async () => {
+    const sequenceTest = new SequenceTest<ActiveStore<CRDTCountTypeRecord>>();
+    sequenceTest.setTestConstructor(() => {
+      DriverFactory.clearRegistrationsForTesting();
+      DriverFactory.register(new MockStorageDriverProvider());
+    
+      const store = new Store('string', Exists.ShouldCreate, null, StorageMode.Direct, CRDTCount);
+      const activeStore = store.activate();
+      return activeStore;
+    });    
+
+    const onProxyMessage = sequenceTest.registerInput('onProxyMessage', 1, 
+      {type: ExpectedResponse.Constant, response: true});
+    const onReceive = sequenceTest.registerInput('onReceive', 1, {type: ExpectedResponse.Void}); 
+
+    const send = sequenceTest.registerOutput('driver.send', 
+      {
+        type: ExpectedResponse.Defer,
+        default: true,
+      }, SequenceOutput.Replace);
+      
+    const idVar = sequenceTest.registerVariable(-1);
+    const isSyncRequest = sequenceTest.registerVariable(false);
+    const model = sequenceTest.registerVariable(() => new CRDTCount(), true);
+  
+    const on = sequenceTest.registerOutput('on',
+      {
+        type: ExpectedResponse.Constant,
+        response: true,
+        onOutput: (value: ProxyMessage<CRDTCountTypeRecord>) => {
+          const syncRequest = sequenceTest.getVariable(isSyncRequest);
+          if (syncRequest) {
+            assert.equal(value.type, ProxyMessageType.ModelUpdate);
+            sequenceTest.setVariable(isSyncRequest, false);
+          } else {
+            if (value.type === ProxyMessageType.Operations) {
+              const currModel: CRDTCount = sequenceTest.getVariable(model);
+              for (const operation of value.operations) {
+                assert.isTrue(currModel.applyOperation(operation));
+              }
+            } else {
+              assert.fail();
+            }
```

Comment:
I think this would be better as:

```
assert.equal(value.type, ProxyMessageType.Operations);
const currModel: CRDTCount = sequenceTest.getVariable(model);
for (const operation of value.operations) {
...
```

It simplifies the control flow and if the code does send the wrong message type, will produce a better failure message.

Reply:
Typescript doesn't know about assert.equal, so this doesn't type check; that's why I've had to do it switched around.

Reply:
Sadly that also doesn't work.